### DISCUSSION TITLE
Question about  inhomogeneity correction: What is otsu?

### DISCUSSION THREAD

**Discussion by geowk** [CoBrALab/RABIES#387] [General]
Hello,

What is otsu and what do the integers 0,1,3,4 indicate?

Is the otsu_thresh used in conjunction with every method option? What if you select method=no_reg, would it still require an integer defined for otsu_thresh?

Thanks.

-g

**Comment by Gab-D-G** [comment]
Hi, 

Sorry for the delays! I was off from work for a few weeks. 

The [otsu method](https://en.wikipedia.org/wiki/Otsu%27s_method) is an image thresholding strategy that finds appropriate thresholds for segragating background from main body of an image. It finds appropriate thresholds based on the histogram of the voxel/pixel intensity distribution (i.e. a valley in the histogram is a good spot to separate two different parts of the image). 

We use otsu thresholding as a data-driven strategy to derive an initial brain mask estimate prior to registration. This step is always run. If no_reg is selected, then no registration step is conducted, and inhomogeneity correction is in fact conducted based on the otsu-generated mask. 

Our script generates 4 different levels of possible otsu threshold (i.e. 4 different cuts in the histogram). So you can select how high in the histogram you want to cut the image by manipulating the threshold. 0 means no threshold is applied, so the mask generated actually covers the whole image.

I hope this helps,
Gabe

**Comment by geowk** [comment]
Thanks for the explanation, Gabe. This was really helpful.

Thanks again.

-g
